### PR TITLE
Firefox Integration

### PR DESCRIPTION
**Pull request by matthewoestreich**
~~and fix issue #14~~ [see here](https://github.com/brumm/tako/pull/16#discussion_r405753393)

### CODE REVIEW

**Comment by brumm** [comment]
Could you fix the conflicts please?

**Comment by matthewoestreich** [comment]
@brumm conflicts have been resolved.

**Comment by matthewoestreich** [comment]
@brumm let me know if there's anything you want changed, etc... Also, I wasn't sure how you wanted things structured so I tried to stick to the existing project structure as much as possible.

I'd be happy to make improvements/changes as you (or anyone, really) see fit.

I've also been using the Firefox build as an temporary Firefox plugin for like 4 or 5 days now and it's been working great!  Love this project 😄

**Comment by brumm** [comment]
Hey! I've been pretty busy the last couple days, and I haven't had time to take a detailed look yet, sorry :) Will get to it as soon as possible!

File: src/api.js [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
+      // A user may have manually removed token from GitHub account,
+      // if they have, this should prompt them for a new token
+      if (response.status === 401 && token) {
+        removeToken().then(() => {
+          throw new Error(
+            `${response.status}: ([tako] It is possible the associated token has been revoked) ${response.statusText}`
+          )
+        })
+      }
+
```
Comment (brumm) [inline src/api.js]: Neat! This should go into a separate PR though, makes it easier to review when it's not packed into a whole bundle of changes. Also, you're removing the token and then throwing an error, did you test this state? I'm not sure whether we can see ErrorBoundary and AskForToken at the same time.

**Comment by brumm** [comment]
@oze4 From the looks of it, there's isn't much (if any?) difference between the two manifest and build processes, correct? I'm wondering why you decided to duplicate the process instead of appending/changing a key in the manifest.

File: src/api.js (hunk 1 above)
Comment (matthewoestreich) [reply src/api.js]: So you want this bit of code removed (for this PR)?  Keep in mind, this code resolves the issue of not being prompted for a new token.

Yes, I did test this state.  If the token is not found, an error is logged to the console and Tako prompts the user for a new token - Tako does not display an error in the UI when the token has been removed, it just prompts for a new token.

**Comment by matthewoestreich** [comment]
@brumm the manifests are nearly identical, that is correct.  I created a nearly identical manifest just to keep the builds separate.  Changing something in the Firefox manifest wouldn't break the Chrome build, etc...  I didn't give it too much thought, and figured it would be easy to change if need be.  But yes, I agree, that could be fine tuned.

Also important to note Firefox manifest and Chrome manifests are not 1:1.  There are some minor differences in how Firefox handles things vs Chrome, that also played into my decision of keeping them separate.

File: src/api.js (hunk 1 above)
Comment (brumm) [reply src/api.js]: >So you want this bit of code removed (for this PR)? Keep in mind, this code resolves the issue of not being prompted for a new token.

Sure! It's a separate topic and a separate fix

Comment (matthewoestreich) [reply src/api.js]: Sounds good. I'll get it removed. If/when this PR gets merged I can submit a separate PR for that if you want me to.  I can make it so the user gets an error stating the token may have been removed. Only after the user has clicked 'Try Again' it will prompt* for a new token. Just an idea...

I can write it however you prefer.

**Comment by matthewoestreich** [comment]
@brumm I went ahead and made those changes.

As far as #14 is concerned, I added an escape hatch for the end-user. Otherwise, if a token is manually removed, Tako is in gridlock and in a state of 'forever error', without providing the user with a way to remedy things.

If you want it in a separate PR, I can remove the 'escape hatch' as well. I just figured it was the least we could do in this PR since it does leave Tako in gridlock w/o giving the user a way out..

**When `Prompt For New Token` is clicked...**
image

**..it clears the token in storage, which ultimately renders `<AskForToken />`**
image

File: src/api.js (hunk 1 above)
Comment (brumm) [reply src/api.js]: Is it possible for you to pull out the changes you needed to do and open an additional PR?
This way, we could get the expired-token-handling in before dealing with all this ;)

Comment (matthewoestreich) [reply src/api.js]: I am confused as to what you want exactly. 

You want me to separate the Firefox changes from the expired token handling changes and submit 2 separate PRs, back to back?....

Comment (brumm) [reply src/api.js]: Yes my dude. This PR is about _Firefox Integration_. Yet here we are spending time on _a different feature, which prevents errors when people revoke their token_. These are two distinct topics and concerns. Had you not put both of those into one PR, we could have gotten the latter one merged already, cause it's a nice and easy change. Instead, it's bunched together with wide-reaching changes that I don't want to merge like this.

Comment (brumm) [reply src/api.js]: There can be multiple open PRs per topic. It's very useful!

Comment (matthewoestreich) [reply src/api.js]: All remnants of the 'expired token handling' have been removed from this PR.

Comment (matthewoestreich) [reply src/api.js]: cc @#20

**Comment by brumm** [comment]
@oze4 Thanks so much for you work on this! I ended up going for low-impact changes to get a compatible FF version out, and thanks to you it went super fast :)

**Comment by matthewoestreich** [comment]
@brumm sure thing. Not really sure why we had this run-around about removing the 'expired token' fix if you weren't going to use any of this...

Lastly, the toolbar icon for Tako is missing in FF.  

Have a good one.

**Comment by brumm** [comment]
Hey dude, no hard feelings. We had the run-around because after like 10 messages you didn't understand that I did not want to mix two distinct concerns into one PR.

After that back-and-forth I was pretty sure that I did not want to hold your hand for a similarly long time through removing 90% of the unnecessary complexity you added here. I'm usually super happy to teach people, but your attitude put me off a bit.

I'm sorry to see that you closed your PRs and issues in a huff, that's not a great look.

I wish you well.
